internal/client/types/custom.go

```diff
@@ -1,15 +1,57 @@
 package types
 
+import "github.com/datarootsio/terraform-provider-dagster/internal/utils"
+
 type CodeLocationsAsDocumentResponse struct {
 	Locations []CodeLocation `json:"locations"`
 }
 
 type CodeLocation struct {
-	Name       string                 `json:"location_name"`
-	Image      string                 `json:"image"`
-	CodeSource CodeLocationCodeSource `json:"code_source"`
+	Name             string                 `json:"location_name"`
+	Image            string                 `json:"image,omitempty"`
+	CodeSource       CodeLocationCodeSource `json:"code_source"`
+	WorkingDirectory string                 `json:"working_directory,omitempty"`
+	ExecutablePath   string                 `json:"executable_path,omitempty"`
+	Attribute        string                 `json:"attribute,omitempty"`
+	Git              CodeLocationGit        `json:"git,omitempty"`
+	AgentQueue       string                 `json:"agent_queue,omitempty"`
+}
+
+func (cl CodeLocation) Validate() error {
```

Comment:
Put this logic in a ConfigValidator in the resource? https://developer.hashicorp.com/terraform/plugin/framework/resources/validate-configuration

That module provides functions like [func ExactlyOneOf(expressions ...path.Expression) resource.ConfigValidator](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.12.0/resourcevalidator#ExactlyOneOf)